UnitTests/CommonTestUtils/SubmoduleTestHelpers.cs

```diff
@@ -11,10 +13,20 @@ public static SubmoduleInfoResult UpdateSubmoduleStructureAndWaitForResult(ISubm
             SubmoduleInfoResult result = null;
             provider.StatusUpdated += Provider_StatusUpdated;
 
-            provider.UpdateSubmodulesStructure(
-                workingDirectory: module.WorkingDir,
-                noBranchText: string.Empty,
-                updateStatus: updateStatus);
+            try
+            {
+                ThreadHelper.JoinableTaskFactory.RunAsync(async () =>
+                {
+                    await provider.UpdateSubmodulesStructureAsync(
+                        workingDirectory: module.WorkingDir,
+                        noBranchText: string.Empty,
+                        updateStatus: updateStatus);
+                });
+            }
+            catch (Exception)
```

Comment:
Do you really expect exceptions here -- especially those that should be ignored silently?
I'd let NUnit catch and log them.
`provider.StatusUpdated -= Provider_StatusUpdated;` should rather be done in a `finally` block.

Reply:
We don't use JTF in tests. We just await.

Reply:
> We don't use JTF in tests. We just await.

The function `UpdateSubmodulesStructure` is not `async`, i.e. no need to `await`. Or have I missed something?
Why do you think it should be called using JTF, @lhiginbotham?

Regarding exception-safety refer to https://github.com/gitextensions/gitextensions/commit/5c803acb86fee03cf3b360b9b7440559577e05ab.

Reply:
I was using JTF to avoiding having to propagate async all the way up to the test level, but I have since learned that it is better to do this async propagation whenever converting sync code to async.